cmd/share-download-main.go

```diff
@@ -89,9 +89,15 @@ func checkShareDownloadSyntax(ctx *cli.Context) {
 		fatalIf(errDummy().Trace(expiry.String()), "Expiry cannot be larger than 7 days.")
 	}
 
-	for _, url := range ctx.Args() {
-		_, _, err := url2Stat(url)
-		fatalIf(err.Trace(url), "Unable to stat `"+url+"`.")
+	// Validate if object exists only if the `--recursive` option was NOT specified
+	isRecursive := ctx.Bool("recursive")
+	if !isRecursive {
+		for _, url := range ctx.Args() {
+			_, _, err := url2Stat(url)
+			if err != nil {
+				fatalIf(errDummy(), "Object `"+url+"` does not exist.")
```

Comment:
url2Stat can send different errors other than just object may not exist. Also doesn't use the traced err. So the origination of the error is lost

To fix the message we need to fix the the odd message is originated inside client-s3.go